frontend/javascripts/oxalis/view/right-menu/advanced_search_popover.js

```diff
@@ -0,0 +1,150 @@
+        datum => datum[this.props.searchKey].toLowerCase().indexOf(searchQuery.toLowerCase()) > -1,
+      );
+      this.selectNextOption();
+    }
+  };
+
+  openSearchPopover = () => {
+    this.setState({ isVisible: true });
+  };
+
+  closeSearchPopover = () => {
+    this.setState({ isVisible: false });
+  };
+
+  render() {
+    const numberOfAvailableOptions = _.values(this.availableOptions).length;
+    const hasNoResults = numberOfAvailableOptions === 0;
+    const hasMultipleResults = numberOfAvailableOptions > 1;
+    const additionalInputStyle = hasNoResults ? { color: "red" } : {};
+    return (
+      <React.Fragment>
+        {this.props.provideShortcut ? (
+          <Shortcut
+            supportInputElements
+            keys="ctrl + shift + f"
+            onTrigger={this.openSearchPopover}
+          />
+        ) : null}
+        <Popover
+          title="Search"
+          trigger="click"
+          placement="rightTop"
+          visible={this.state.isVisible}
+          mouseLeaveDelay={10}
+          onVisibleChange={isVisible =>
+            isVisible ? this.openSearchPopover() : this.closeSearchPopover()
+          }
+          content={
+            // Only render search components when the popover is visible
+            // This ensures that the component is completely re-mounted when
+            // the popover is opened. Thus unnecessary computations are avoided.
+            this.state.isVisible && (
+              <React.Fragment>
+                <Shortcut supportInputElements keys="escape" onTrigger={this.closeSearchPopover} />
+                <InputGroup compact style={{ width: 450 }}>
+                  <Input
+                    style={{ width: "calc(100% - 100px)", ...additionalInputStyle }}
+                    value={this.state.searchQuery}
+                    placeholder="Enter your search keywords"
+                    onPressEnter={this.selectNextOption}
+                    onChange={evt => this.onSearchQueryChanged(evt.target.value)}
+                    autoFocus
+                  />
+                  <Tooltip title="Previous">
+                    <ButtonComponent
+                      style={{ width: 50 }}
+                      onClick={this.selectPreviousOption}
+                      disabled={!hasMultipleResults}
+                    >
+                      <Icon type="up" />
+                    </ButtonComponent>
+                  </Tooltip>
+                  <Tooltip title="Next">
+                    <ButtonComponent
+                      style={{ width: 50 }}
+                      onClick={this.selectNextOption}
+                      disabled={!hasMultipleResults}
+                    >
+                      <Icon type="down" />
+                    </ButtonComponent>
+                  </Tooltip>
+                </InputGroup>
+                {!hasNoResults ? (
+                  <div style={{ paddingLeft: 4, paddingTop: 8 }}>
```

Comment:
The popover would be more concise if displaying the number of matches wouldn't take an extra line. I like the way chrome does this, but that doesn't seem easy to implement. Maybe add it to the right (without the `matches`, simply x/y), or to the title. I'm not sure what looks best.